dotnet/src/SemanticKernel.Abstractions/Functions/ReadOnlyKernelPlugin.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+#pragma warning disable CA1716 // Identifiers should not match keywords
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Represents a plugin that may be registered with a <see cref="Kernel"/>.</summary>
+/// <remarks>
+/// A plugin is named read-only collection of functions. There is a many-to-many relationship between
+/// plugins and functions: a plugin may contain any number of functions, and a function may
+/// exist in any number of plugins.
+/// </remarks>
+public abstract class ReadOnlyKernelPlugin : IEnumerable<KernelFunction>
```

Comment:
THis class can be extended to not be readonly, I would avoid setting this as a `ReadONly` in the name, maybe `KernelPluginBase` ?

Reply:
Using `KernelPlugin`